src/proto/single/ent/progress.rs

```diff
@@ -38,6 +38,12 @@ pub struct ProgressUpdate {
     pub reserve_until: Option<DateTime<Utc>>,
 }
 
+impl AsRef<Self> for ProgressUpdate {
```

Comment:
Reading <https://doc.rust-lang.org/std/convert/trait.AsRef.html#relation-to-borrow>, I'm starting to wonder if we should prefer `Borrow` over `AsRef` for the functions that just want a reference. That would also (I believe) mean this `impl` could go away.

Reply:
Well, I this would go away indeed and being Borrow will do the job internally. I guess the question is, how ergonomic and expected this is. Here is how the `get_progress_ would look like:
```rust
pub async fn get_progress<J>(&mut self, jid: J) -> Result<Option<Progress>, Error>
    where
        J: Borrow<JobId> + Sync,
    {
        let cmd = FetchProgress::new(jid.borrow());
        self.issue(&cmd).await?.read_json().await
    }
```

Say I've got a struct keeping JobId and and TraceId, for which the AsRef<JobId> is implemented. It would be nice if I could pass it to `Client::get_progress` which is not possible due to:
```rust
```

Reply:
Oh, right, `Borrow` is overloaded to also mean equivalence for `Hash` and friends. That's awkward. Okay, then I'm fine with sticking with `AsRef` :+1: